lib/CompositeController.js

```diff
@@ -101,7 +149,7 @@ module.exports = class CompositeController extends BaseController {
     if (!objectPath.has(child, ['metadata', 'namespace']) && krm.namespaced) {
       objectPath.set(child, ['metadata', 'namespace'], this.namespace);
     }
-    let childUri = krm.uri({ name: child.metadata.name, namespace: child.metadata.namespace });
+    childUri = krm.uri({ name: childName, namespace: objectPath.get(child, 'metadata.namespace') });
```

Comment:
```suggestion
    childUri = krm.uri({ name: childName, namespace: objectPath.get(child, 'metadata.namespace','default') });
```

Reply:
no i want this to be undefined if it is truely undefined